src/fusor/translator.py

```diff
@@ -0,0 +1,580 @@
+"""Module for translating output from fusion detection methods to FUSOR AssayedFusion
+objects
+"""
+
+import logging
+from enum import Enum
+
+import polars as pl
+from cool_seq_tool.schemas import Assembly
+
+from fusor.fusor import FUSOR
+from fusor.models import (
+    Assay,
+    AssayedFusion,
+    CausativeEvent,
+    EventType,
+    GeneElement,
+    TranscriptSegmentElement,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class Caller(str, Enum):
+    """Define different supported callers"""
+
+    JAFFA = "JAFFA"
+    STAR_FUSION = "STAR-Fusion"
+    FUSION_CATCHER = "FusionCatcher"
+    FUSION_MAP = "FusionMap"
+    ARRIBA = "Arriba"
+    CICERO = "CICERO"
+    MAPSPLICE = "MapSplice"
+    ENFUSION = "EnFusion"
+    GENIE = "GENIE"
+
+
+class Translator:
+    """Class for translating outputs from different fusion detection algorithms
+    to FUSOR AssayedFusion objects
+    """
+
+    def __init__(self, fusor: FUSOR) -> None:
+        """Initialize Translator class
+
+        :param fusor: A FUSOR instance
+        """
+        self.fusor = fusor
+
+    def _format_fusion(
+        self,
+        gene_5prime: GeneElement,
+        gene_3prime: GeneElement,
+        tr_5prime: TranscriptSegmentElement | None = None,
+        tr_3prime: TranscriptSegmentElement | None = None,
+        ce: CausativeEvent | None = None,
+        rf: bool | None = None,
+        assay: Assay | None = None,
+    ) -> AssayedFusion:
+        """Format classes to create AssayedFusion objects
+
+        :param gene_5prime: 5'prime GeneElement
+        :param gene_3prime: 3'prime GeneElement
+        :param Optional[tr_5prime]: 5'prime TranscriptSegmentElement
+        :param Optional[tr_3prime]: 3'prime TranscriptSegmentElement
+        :param Optional[ce]: CausativeEvent
+        :param Optional[rf]: A boolean indicating if the reading frame is preserved
+        :param Optional[assay]: Assay
```

Comment:
you don't need to put this `Optional[]` stuff in here. just `:param name_of_arg: description of arg`